### DISCUSSION TITLE
What is the relationship between the `@prisma/client` node module and `.prisma/client`

### DISCUSSION THREAD

**Discussion by campbellwmorgan** [prisma/prisma#9550] [Q&A] [answered]
I am having a lot of problems with using parcel in monorepo that uses some types from my prisma client in one of the imported packages.

Parcel seems to be struggling with this part

```
Server running at http://localhost:1234 
🚨  /home/user/project/node_modules/.prisma/client/index-browser.js:6:12: Cannot resolve dependency '@prisma/client/runtime/index-browser'
  4 | const {
  5 |   Decimal
> 6 | } = require('@prisma/client/runtime/index-browser')
    |            ^
  7 | 
  8 | 
  9 | const Prisma = {}
  ```
  
  And I wondered, if somebody could explain the relationship between `@prisma/client` and `.prisma/client`? This is also interesting to me because VSCode frequently auto-imports `.prisma/client` which tends to cause build problems in other pipelines.

Many thanks in advance

**Comment by ryands17** [comment]
@campbellwmorgan 👋 

We have a [section](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client#the-location-of-prisma-client) in the docs of why we have opted for this convention. I don't know about the workings of Parcel, but you could try generating the client using a custom output path and then check if that works 🙂

**Reply by campbellwmorgan** [reply]
@ryands17 thanks a lot for the reply! Yes somewhat useful -- but sadly has not solved my parcel problem :( 

In the end I had to do a very long-winded workaround involving some scripts to create a new workspace package from the `index-browser.js` file and the browser `index-browser.d.ts` . If anybody else runs into this problem, i'd be happy to share the files of the workaround here

**Reply by ryands17** [reply]
I would suggest sharing it here so that it's a great reference for others who stumble upon this problem 🙂

**Comment by campbellwmorgan** [comment, marked as answer]
My solution was to create a new workspace `package` called `prisma-browser` in my monorepo and use `esbuild` to bundle prisma client correctly and then a script to copy the typescript type declaration to my dist folder.

My package json has the following sections
```json
"scripts": {
    "copy-prisma-types": "node scripts/copyPrismaTypes",
    "bundle-prisma": "yarn esbuild --outfile=dist/index.js --format=cjs --bundle --tree-shaking=true src/index.ts",
    "build": "yarn bundle-prisma && yarn copy-prisma-types"
  },
  "devDependencies": {
    "@prisma/client": "3.0.1",
    "esbuild": "^0.13.3"
  }
 ```
 Where `src/index.ts` looks like 
 ```typescript
 export * from "@prisma/client";
```
And `scripts/copyPrismaTypes.js` looks like
```javascript
/**
 * Copies the prisma type declaration to this dist folder
 * - Goal is to make a browser-only compatible distribution that doesn't cause build problems
 *  so we can share types frontend/backend
 */

const fs = require('fs');
const path = require('path');

console.log("Attempting to copy prisma type declarations...");
const filePath = require.resolve('.prisma/client/index.d.ts');

const contents = fs.readFileSync(filePath);

const outputName = path.join(__dirname, '/../dist/index.d.ts');

fs.writeFileSync(outputName, contents);

console.log("Prisma type declarations copied to", outputName);

```

You may ask why I am using `esbuild` and `parcel` and the answer is that many of the modules in my app are not compatible with esbuild

**Reply by azertyalex** [reply]
I'm getting following error when applying your structure to my prisma shared client
```
"UserRoleValue" is not exported by "../../packages/database/dist/index.js",
```
I seem to be missing something?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
